lib/restify-plugins/curl-user-agent.js

```diff
@@ -0,0 +1,60 @@
+// Copyright 2012 Mark Cavage, Inc.  All rights reserved.
+// Copyright (c) 2016, Joyent, Inc.
+
+// This is basically the user agent pre-route handler from
+// https://github.com/restify/plugins with a bug fix that was submitted upstream
+// at https://github.com/restify/plugins/pull/40. If and when
+// https://github.com/restify/plugins/pull/40 is merged, the goal is to backport
+// it to whatever restify branch VMAPI uses at that time and remove this local
+// copy.
+
+var assert = require('assert-plus');
+
+/**
+ * This basically exists for curl.  curl on HEAD requests usually
+ * just sits there and hangs, unless you explicitly set
+ * Connection:close.  And in general, you probably want to set
+ * Connection: close to curl anyway.
```

Comment:
> Would it make sense to add a test that explicitly ensures things still work fine when the curl client has passed 'Connection: close'

The `connection` header sent by VMAPI will be overriden to `close` only if the client has a user agent that matches `curl`. If a client uses `curl` and set the `connection` header to `close`, then the server will send a response with a `close` header, which seems fine.

However I agree it wouldn't hurt to add more test cases that send different `connection` header values when using `curl` and non-curl clients.

> Is there any other possible value for 'Connection: ' that a client might ever set that we're silently overriding? Does that matter?

Clients can send other values for the `connection` header, like `upgrade`. VMAPI doesn't support that though for now as far as I know, so I would not add a test case for this.